File: src/_03_BehavioralPattern/_03_02_Command/LightOffCommand.js
import Command from "./Command.js";
import LightOnCommand from "./LightOnCommand.js";

/**
 * @typedef { import("./Light").default } Light
 */

export default class LightOffCommand extends Command {
  /** @type { Light } */
  #light;

  /** @param { Light } light */
  constructor(light) {
    super();
    this.#light = light;
  }

  /** @override */
  execute() {
    this.#light.off();
  }

  /** @override */
  undo() {
    new LightOnCommand(this.#light).execute();
  }
}

File: src/_03_BehavioralPattern/_03_02_Command/java/after/LightOffCommand.java
package _03_BehavioralPattern._03_02_Command.java.after;

public class LightOffCommand implements Command {

  private Light light;

  public LightOffCommand(Light light) {
    this.light = light;
  }

  @Override
  public void execute() {
    this.light.off();
  }

  @Override
  public void undo() {
    new LightOnCommand(this.light).execute();
  }
  
}
